### PR TITLE
Run 198 Make the correct arrangement of elements on all types of screens in onboarding

### PR DESCRIPTION
**Pull request by Kengii**
## Description 📝

Make the correct arrangement of elements on all types of screens in onboarding

---

### Tests ✅

Simulator: Iphone SE, 6s, SE gen 3, 8 Plus, 11 pro, 11 pro max, 14 pro

### CODE REVIEW

**Review by Qeezii — APPROVED** [review]

**Comment by FilyutaDima** [comment]
Incorrect design of button "Start" on the Library screen. You have to change the background color and title color.

Current design:
Снимок экрана 2023-01-14 в 19 10 03

Figma:
Снимок экрана 2023-01-14 в 19 10 33

**Review by arti1233 — APPROVED** [review]

**Review by FilyutaDima — APPROVED** [review]

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenCell.swift [hunk 1]
```diff
@@ -62,9 +62,18 @@ class OnboardingScreenCell: UICollectionViewCell {
         contentView.addSubview(onboardingDescription)
         onboardingDescription.snp.makeConstraints { make in
             make.top.equalTo(onboardingSlideTitle.snp.bottom).offset(32)
-            make.width.equalTo(230)
             make.height.equalTo(70)
             make.centerX.equalTo(contentView)
+            if !DeviceType.iPhoneSE && !DeviceType.isIphone78Plus{
+                make.trailing.equalToSuperview().inset(64)
+                make.leading.equalToSuperview().inset(64)
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenCell.swift]: redundant line of code. Please, read the doc for SnapKit.

Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenCell.swift]: Please, think about constants.

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenCell.swift [hunk 2]
```diff
@@ -62,9 +62,18 @@ class OnboardingScreenCell: UICollectionViewCell {
...
+            } else if !DeviceType.iPhoneSE {
+                make.trailing.equalToSuperview().inset(72)
+                make.leading.equalToSuperview().inset(72)
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenCell.swift]: Please, remove copy-paste.

**Review by AliaksandrNiehaj — CHANGES_REQUESTED** [review]

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift [hunk 3]
```diff
@@ -106,18 +108,43 @@ class OnboardingView: UIView {
...
+        }
+    }
+    
+    func configureOnboardingStartButton() {
+        
+        onboardingStartButton.isHidden = true
+        
+        let radiusConstant: CGFloat = DeviceType.iPhoneSE ? 6.58 : 8
+        let borderConstant: CGFloat = DeviceType.iPhoneSE ? 0.82 : 1
+        let fontConstant: CGFloat = DeviceType.iPhoneSE ? 24 : 30
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift]: Could you, please, explain, what does it mean: 6.58, 8, 0.82, 1?

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift [hunk 4]
```diff
@@ -106,18 +108,43 @@ class OnboardingView: UIView {
...
+        onboardingStartButton.layer.cornerRadius = radiusConstant
+        onboardingStartButton.layer.borderWidth = borderConstant
+        onboardingStartButton.setTitle(L10n.Onboarding.start, for: .normal)
+        onboardingStartButton.backgroundColor = UIColor(red: 0.937, green: 0.804, blue: 0.576, alpha: 1)
+        onboardingStartButton.layer.borderColor = UIColor(red: 1, green: 0.917, blue: 0.792, alpha: 0.7).cgColor
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift]: Use, please, constants.

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift [hunk 5]
```diff
@@ -106,18 +108,43 @@ class OnboardingView: UIView {
...
+        onboardingStartButton.translatesAutoresizingMaskIntoConstraints = false
+        onboardingStartButton.titleLabel?.font = FontFamily.AmaticSC.bold.font(size: fontConstant)
+        onboardingStartButton.setTitleColor(UIColor(red: 0.825, green: 0.77, blue: 0.677, alpha: 1), for: .highlighted)
+        onboardingStartButton.setTitleColor(UIColor(red: 0.294, green: 0.282, blue: 0.259, alpha: 1), for: .normal)
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift]: The same (constants).

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift [hunk 6]
```diff
@@ -106,18 +108,43 @@ class OnboardingView: UIView {
...
+        addSubview(onboardingStartButton)
+        onboardingStartButton.addTarget(self, action: #selector(goToNextSlide), for: .touchUpInside)
+        onboardingStartButton.addTarget(self, action: #selector(changeButtonColor(_:)), for: .touchDown)
+        onboardingStartButton.addTarget(self, action: #selector(buttonNormalColor(_:)), for: .touchUpInside)
+        onboardingStartButton.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.width.equalTo(264)
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift]: Fix hardcode, please

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift [hunk 7]
```diff
@@ -106,18 +108,43 @@ class OnboardingView: UIView {
...
+            make.height.equalTo(48)
+            let factor: CGFloat = DeviceType.iPhoneSE ? 0.85 : 0.78
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift]: Could you, please, explain, what does it mean: 0.85, 0.78?

**Review by AliaksandrNiehaj — CHANGES_REQUESTED** [review]

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenCell.swift [hunk 8]
```diff
@@ -62,9 +62,11 @@ class OnboardingScreenCell: UICollectionViewCell {
         contentView.addSubview(onboardingDescription)
         onboardingDescription.snp.makeConstraints { make in
             make.top.equalTo(onboardingSlideTitle.snp.bottom).offset(32)
-            make.width.equalTo(230)
             make.height.equalTo(70)
             make.centerX.equalTo(contentView)
+            let constant: CGFloat = DeviceType.iPhoneSE ? (viewWidth / 7.6) : (viewWidth / 5.45)
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenCell.swift]: @Kengii what does it mean 7.6 and 5.45?
It looks like Kalashnikov machine gun cartridge calibers.

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenCell.swift [hunk 9]
```diff
@@ -62,9 +62,11 @@ class OnboardingScreenCell: UICollectionViewCell {
...
+            make.trailing.equalToSuperview().inset(constant)
+            make.leading.equalToSuperview().inset(constant)
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenCell.swift]: copy paste

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenVC.swift [hunk 10]
```diff
@@ -17,8 +17,12 @@ class OnboardingScreenVC: UIViewController {
             contentView.pageControll.currentPage = currentPage
             contentView.configureCustomPageControll()
             if currentPage == onboardingModel.count - 1 {
-                contentView.onboardingNextSlideButton.setTitle(L10n.Onboarding.start, for: .normal)
+                contentView.onboardingNextSlideButton.isHidden = true
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenVC.swift:20]: think, please, about ternary operator.

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift [hunk 11]
```diff
@@ -106,18 +108,43 @@ class OnboardingView: UIView {
...
-            } else {
-                make.bottom.equalToSuperview().inset(130)
-                make.centerX.equalToSuperview()
-                make.width.equalTo(264)
-                make.height.equalTo(48)
-            }
+            make.centerX.equalToSuperview()
+            make.width.equalTo(264)
+            make.height.equalTo(48)
+            let factor: CGFloat = DeviceType.iPhoneSE ? 0.85 : 0.78
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift]: Fix, please, 0.85/0.78

**Review by AliaksandrNiehaj — CHANGES_REQUESTED** [review]

**Comment by Kengii** [comment]
SE
SE
SE 3
SE 3
13 mini
13 mini
8 plus
8 Plus
X
X
14

14 pro max
14 pro max

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift [hunk 12]
```diff
@@ -91,33 +99,49 @@ class OnboardingView: UIView {
...
+        
+        onboardingStartButton.isHidden = true
+        onboardingStartButton.layer.cornerRadius = 8
+        onboardingStartButton.layer.borderWidth = 1
+        onboardingStartButton.setTitle(L10n.Onboarding.start, for: .normal)
+        onboardingStartButton.backgroundColor = UIColor(red: 0.937, green: 0.804, blue: 0.576, alpha: 1)
+        onboardingStartButton.layer.borderColor = UIColor(red: 1, green: 0.917, blue: 0.792, alpha: 0.7).cgColor
+        onboardingStartButton.translatesAutoresizingMaskIntoConstraints = false
+        onboardingStartButton.titleLabel?.font = FontFamily.AmaticSC.bold.font(size: 24)
+        onboardingStartButton.setTitleColor(UIColor(red: 0.825, green: 0.77, blue: 0.677, alpha: 1), for: .highlighted)
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift]: Use, please, introduced colors.

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift [hunk 13]
```diff
@@ -91,33 +99,49 @@ class OnboardingView: UIView {
...
+        onboardingStartButton.setTitleColor(UIColor(red: 0.294, green: 0.282, blue: 0.259, alpha: 1), for: .normal)
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift]: Use, please, introduced colors.

**Review by AliaksandrNiehaj — CHANGES_REQUESTED** [review]

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenCell.swift [hunk 14]
```diff
@@ -13,6 +13,10 @@ class OnboardingScreenCell: UICollectionViewCell {
     private let onboardingDescription = UILabel()
     private let onboardingImageView = UIImageView()
     
+    // MARK: Constants
+    let constantConstraintsDescription: CGFloat = DeviceType.iPhoneSE ? (viewWidth / 7.6) : (viewWidth / 5.45)
+    // this is a constant, a divider that is calculated from the screen dimensions, for the correct placement of the Description on the screen
+
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenCell.swift:18]: Comment must be written above the var/let. Please, fix.

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenCell.swift [hunk 15]
```diff
@@ -62,9 +66,10 @@ class OnboardingScreenCell: UICollectionViewCell {
         contentView.addSubview(onboardingDescription)
         onboardingDescription.snp.makeConstraints { make in
             make.top.equalTo(onboardingSlideTitle.snp.bottom).offset(32)
-            make.width.equalTo(230)
             make.height.equalTo(70)
             make.centerX.equalTo(contentView)
+            make.trailing.equalToSuperview().inset(constantConstraintsDescription)
+            make.leading.equalToSuperview().inset(constantConstraintsDescription)
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenCell.swift]: Copy paste.

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenVC.swift (hunk 10 above)
Comment (AliaksandrNiehaj) [reply Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingScreenVC.swift:20]: ?

File: Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift [hunk 16]
```diff
@@ -17,10 +17,17 @@ class OnboardingView: UIView {
...
+    let onboardingNextSlideButton = UIButton()
+    let onboardingStartButton = UIButton()
     let pageControllStackView = UIStackView()
     let pageControll = CustomPageControll()
     
+    // MARK: Constants
+    let factorConstant: CGFloat = DeviceType.iPhoneSE ? 0.135 : 0.207
+    let heightButton: CGFloat = 48
+    let widthButton: CGFloat = 264
+    // this is a constant, a multiplier that is calculated from the screen dimensions, for the correct placement of the button on the screen
```
Comment (AliaksandrNiehaj) [inline Runar/Runar/Controllers/Misc/OnboardingScreen/OnboardingView.swift]: The same about comment.

**Review by AliaksandrNiehaj — APPROVED** [review]

**Pull request merged**